model_analyzer/config/generate/base_model_config_generator.py

```diff
@@ -212,6 +199,38 @@ def make_model_config(
 
         return model_config
 
+    @staticmethod
+    def make_ensemble_model_config(
+            model: ModelProfileSpec, ensemble_submodels: List[ModelConfig],
+            model_variant_name_manager: ModelVariantNameManager) -> ModelConfig:
+        """
+        Loads the base model config from the model repository, and then mutates
+        the names to match the ensemble submodels
+        """
+        model_name = model.model_name()
+        model_config_dict = model.get_default_config()
+        ensemble_config_dicts = [
+            model.to_dict() for model in ensemble_submodels
+        ]
+        ensemble_key = ModelVariantNameManager.make_ensemble_submodel_key(
+            ensemble_config_dicts)
+
+        (variant_found, variant_name
+        ) = model_variant_name_manager.get_ensemble_model_variant_name(
+            model_name, ensemble_key)
+
+        model_config_dict['name'] = variant_name
+        model_config = ModelConfig.create_from_dictionary(model_config_dict)
+
+        for submodel in ensemble_submodels:
+            variant_name = submodel.get_field("name")
+            submodel_name = variant_name[:variant_name.find("_config_")]
```

Comment:
can you create a static helper function for this?  I think result_manager.py line 155 would want to use the same.

Reply:
Done